test/Microsoft.ML.Tests/OnnxConversionTest.cs

```diff
@@ -2146,6 +2146,80 @@ public void NonDefaultColNamesMultiClassificationOnnxConversionTest()
             Done();
         }
 
+        [Fact]
+        public void PcaOnnxConversionWithCustomOpsetVersionTest()
+        {
+            var dataSource = GetDataPath(TestDatasets.generatedRegressionDataset.trainFilename);
+
+            var mlContext = new MLContext(seed: 1);
+            var dataView = mlContext.Data.LoadFromTextFile(dataSource, new[] {
+                new TextLoader.Column("label", DataKind.Single, 11),
+                new TextLoader.Column("features", DataKind.Single, 0, 10)
+            }, hasHeader: true, separatorChar: ';');
+
+            bool[] zeroMeans = { true, false };
+            foreach (var zeroMean in zeroMeans)
+            {
+                var pipeline = ML.Transforms.ProjectToPrincipalComponents("pca", "features", rank: 5, seed: 1, ensureZeroMean: zeroMean);
+                var model = pipeline.Fit(dataView);
+                var transformedData = model.Transform(dataView);
+                var onnxModel = mlContext.Model.ConvertToOnnxProtobuf(model, dataView, 9);
+
+                var onnxFileName = "pca.onnx";
+                var onnxModelPath = GetOutputPath(onnxFileName);
+
+                SaveOnnxModel(onnxModel, onnxModelPath, null);
+
+                if (IsOnnxRuntimeSupported())
+                {
+                    // Evaluate the saved ONNX model using the data used to train the ML.NET pipeline.
+                    var onnxEstimator = mlContext.Transforms.ApplyOnnxModel(onnxModelPath);
+                    var onnxTransformer = onnxEstimator.Fit(dataView);
+                    var onnxResult = onnxTransformer.Transform(dataView);
+                    CompareSelectedColumns<float>("pca", "pca", transformedData, onnxResult);
+                }
+            }
+            
+          Done();
+        }
+        
+        [Fact]
+        public void OneHotHashEncodingOnnxConversionWithCustomOpSetVersionTest()
```

Comment:
>OneHotHashEncodingOnnxConversionWithCustomOpSetVersionTest [](start = 20, length = 58)

Is there also a sample for ONNX conversion? If yes, I would add an example there for how to pass a custom opset version.

Reply:
I've just noticed there's no sample about Onnx conversion in
[https://github.com/dotnet/machinelearning/tree/master/docs/samples/Microsoft.ML.Samples](https://github.com/dotnet/machinelearning/tree/master/docs/samples/Microsoft.ML.Samples)

Would it be a good idea to add one on this PR?

Also, BTW, I'd add a new test on [Functional Tests](https://github.com/dotnet/machinelearning/blob/master/test/Microsoft.ML.Functional.Tests/ONNX.cs) (or perhaps modify the tests you have in here) to test the new public overload that was created for `ConverToOnnx`. Currently, the new tests added on this PR directly call the new ConvertToOnnxProtobuf internal API, and I think it would be better to also test the public API, to make sure that future changes don't break that API.

---
In reply to: [432950286](https://github.com/dotnet/machinelearning/pull/5176#discussion_r432950286) [](ancestors = 432950286)

Reply:
onnx conversion sample will be added in another PR as this PR targets support onnx export for previous opset version